src/ComposerRequireChecker/FileLocator/LocateComposerPackageSourceFiles.php

```diff
@@ -27,11 +27,14 @@ public function __invoke(string $composerJsonPath) : Generator
 
     private function getFilePaths(array $sourceDirs, string $packageDir) : array
     {
+        $flattened = array_reduce($sourceDirs, function(array $sourceDirs, $sourceDir) {
+            return $sourceDirs + (is_array($sourceDir) ? $sourceDir : [$sourceDir]);
```

Comment:
> you can cast to an array?

not always safely, which is why I asked for the types here